### PR TITLE
✨ feat: Add unique constraint and index to `code` field in `BaseLocation` model

### PR DESCRIPTION
**Pull request by ARYAN-NIKNEZHAD**
### Summary

This pull request adds a unique constraint and index to the `code` field in the `BaseLocation` model. This ensures that each location has a unique code and improves query performance on the `code` field.

### Changes

- Added `unique=True` to the `code` field in the `BaseLocation` model.
- Added an index to the `code` field to enhance performance.
- Updated the database schema to reflect these changes.

### Motivation

The unique constraint ensures data integrity by preventing duplicate codes for locations. The index on the `code` field will significantly improve the performance of queries that filter or search by code, especially for large datasets.

### Related Issues

This change addresses the need for unique location codes and performance improvements in queries involving the `code` field.

### Notes

Please review the changes and provide feedback. If there are any concerns or suggestions for further improvements, feel free to modify as needed.

### CODE REVIEW

**Comment by ARYAN-NIKNEZHAD** [comment]
This pull request closes #9. 🚀

File: iranian_cities/models.py [hunk 1]
```diff
@@ -1,160 +1,181 @@
...
-    class Meta:
+class Province(BaseLocation):
+    """
+    Represents a province entity within the application.
+    """
+    class Meta(BaseLocation.Meta):
         verbose_name = _('Province')
         verbose_name_plural = _('Provinces')
-        ordering = ('id',)
+        db_table_comment = _("Model representing a province.")
```
Comment (sepehr-akbarzadeh) [inline iranian_cities/models.py]: db_table_comment must not have gettext lazy. remove it.
also add table name to sage_iranian_cities

**Review by sepehr-akbarzadeh — CHANGES_REQUESTED** [review]
db_table_comment must not have gettext lazy. remove it.
also add table name to sage_iranian_cities

File: iranian_cities/models.py (hunk 1 above)
Comment (ARYAN-NIKNEZHAD) [reply iranian_cities/models.py]: Hi @sepehr-akbarzadeh ,

Thank you for your feedback! 😊

I’ve made the following changes based on your suggestions:

Removed gettext_lazy from db_table_comment to ensure proper handling of database comments.
Added the table name to sage_table_name for clarity and consistency.

Please review the changes and let me know if there are any further adjustments needed. Your feedback is highly 
appreciated!

**Pull request merged**